[TensorExpr] Wrap fuser in a class.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #42938 [TensorExpr] Cleanup logic in the TensorExpr fuser pass.
* #42937 [TensorExpr] Remove redundant checks from canHandle in TE fuser.
* **#42936 [TensorExpr] Wrap fuser in a class.**

Differential Revision: [D23084407](https://our.internmc.facebook.com/intern/diff/D23084407)

Commit:
[TensorExpr] Wrap fuser in a class.

[ghstack-poisoned]

eellison commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -188,180 +189,210 @@ bool canHandle(Node* node, AliasDb& aliasDb) {
+ private:
+  // Find prim::profile nodes in the graph, extract value -> type information
+  // from it, and in the end remove the nodes from the graph.
+  void constructTypeInfoMap() {
```
it would be nice to keep this PR to only code movement

eellison reviewed (approved):
I think this is just code movement, so accepting. It would be nice to remove the couple of newly added function calls (that don't do anything) from this PR

Commit:
Update on "[TensorExpr] Wrap fuser in a class."


Differential Revision: [D23084407](https://our.internmc.facebook.com/intern/diff/D23084407)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Wrap fuser in a class."


Differential Revision: [D23084407](https://our.internmc.facebook.com/intern/diff/D23084407)

[ghstack-poisoned]
